### ISSUE TITLE
stats: locks at start up

### ISSUE THREAD

**Issue by nicolagi** [9fans/plan9port#349]
Very simply, running `stats` doesn't do anything, just blocks.

I have a syscall trace for this: https://gist.github.com/nicolagi/2e7c0f175e3b9cca346f18e3427b2741

Took a while to repro wrapping the program with strace, which suggests a concurrency related bug.

By the last two lines, stats seems to be blocking while trying to read fro auxstats.

```
26347 read(3, "ork 1014917 1000\nfault 286332201"..., 255) = 40
26347 read(3, 
```

**Comment by dancrossnyc** [comment]
It's actually hanging on `_displayinit()`, in `muxrpc`. I haven't had time to track it further, but it's somewhere in there. Note that `_displayinit` has a note that says something along the lines of, "this is the only place we can reschedule."

Concurrency bug is certainly on the right track.

**Comment by dancrossnyc** [comment]
Ok. It's block in `drawrecv` on `readn` from d->srvfd, trying to read a 9P message from drawsrv. This implies that that message is not coming.

**Comment by dancrossnyc** [comment]
(Er, rather, reading from the pipe from `devdraw`)

**Comment by dancrossnyc** [comment]
Hmm. Interesting. The `putenv` call in `_displayconnect` is hanging.

**Comment by dancrossnyc** [comment]
Ah...and now I've found it. It's actually hanging in `p9malloc`.

**Comment by dancrossnyc** [comment]
Fixed in 7bf2db4c2ae30c0f7b320e57060715bf6279e98a.
